### PR TITLE
add 'alter' function to DA.Map

### PR DESCRIPTION
**Pull request by S11001001**
Suggested in #7365, moving the `alter` function from the triggers library to `DA.Map`.

* [x] merge #7425, remove from DA.Trigger.Internal

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-stdlib-src/DA/Map.daml [hunk 1]
```diff
@@ -157,6 +158,25 @@ delete = primitive @"BEGenMapDelete"
...
 insert = primitive @"BEGenMapInsert"
 
+-- | Update the value in 'm' at 'k' with 'f', inserting or deleting as
+-- required.  'f' will be called with either the value at 'k', or `None`
+-- if absent; 'f' can return `Some` with a new value to be inserted in
+-- 'm' (replacing the old value if there was one), or `None` to remove
+-- any 'k' association 'm' may have.
+--
+-- Some implications of this behavior:
+--   alter identity k = identity
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Map.daml:168]: I think you need an extra line so it gets interpreted as code block:

```suggestion
-- Some implications of this behavior:
-- 
--   alter identity k = identity
```

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you! I added some comments on the formatting of the documentation.

File: compiler/damlc/daml-stdlib-src/DA/Map.daml [hunk 2]
```diff
@@ -157,6 +158,25 @@ delete = primitive @"BEGenMapDelete"
 insert : Ord k => k -> v -> Map k v -> Map k v
 insert = primitive @"BEGenMapInsert"
 
+-- | Update the value in 'm' at 'k' with 'f', inserting or deleting as
```
[the remark this answers is not in the document]
Comment (S11001001) [reply compiler/damlc/daml-stdlib-src/DA/Map.daml]: Ah, I see. Thanks. 400ef896d0dd80c9a6c7df24da6bde2cccb3852f

File: compiler/damlc/daml-stdlib-src/DA/Map.daml (hunk 1 above)
Comment (S11001001) [reply compiler/damlc/daml-stdlib-src/DA/Map.daml]: Thanks. 4751c9271a3a4b84f199ea72dae97e7eeae92900

**Review by cocreature — COMMENTED** [review]
Thank you!

**Comment by hurryabit** [comment]
Nice. Merci.

**Pull request merged**